rust/arrow/src/array/equal/structure.rs

```diff
@@ -15,13 +15,15 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use crate::array::ArrayData;
+use crate::{array::data::count_nulls, array::ArrayData, buffer::Buffer};
 
 use super::equal_range;
 
 fn equal_values(
```

Comment:
Given the number of parameters this function has, it might be worth adding doc comments on it (especially as `lhs_nulls` and `rhs_nulls` now have non trivial semantics (nullability is checked by `AND`ing them together)

Reply:
I've added the details on the same function in `crate::array::equal::mod`. I've copied the same doc from the function in `mod` to here, but I think it's redundant as the functions are private, as long as the `mod` one is sufficiently documented.